lib/context.js

```diff
@@ -6,25 +6,40 @@
 'use strict'
 
 const util = require('util')
+const semver = require('semver')
+
+// Version middleware is stable
+// See: https://nextjs.org/docs/advanced-features/middleware
+const MIN_SUPPORTED_VERSION = '12.2.0'
 
 module.exports = function initialize(shim, ctx) {
+  const nextVersion = shim.require('./package.json').version
+
+  if (semver.lt(nextVersion, MIN_SUPPORTED_VERSION)) {
+    shim.logger.debug(
+      'Next.js middleware instrumentation only supported on >=12.2.0, got %s',
+      nextVersion
+    )
+    return
+  }
+
```

Comment:
While I understand the motivation to stop supporting versions earlier than 12.2, I wonder if our users are getting enough of a heads-up that this is the big version where stability happened. It seems odd that it happens in a minor Next.js version.

Checking my understanding: we still support next.js previous to 12.1, just not its middleware, and we also no longer check if `getModuleContext` is async or not because that only matters in the context of middleware.

Reply:
we're still in 0.x of our package because it's still unofficially experimental for us.  We added support for a feature Next.js middleware that was experimental for the project.  So pre 12.2.0 we will still track server calls and api calls but just not middleware.  And yes we stopped checking if getModuleContext is async because it always is 12.2.0+